common/tools/dev-tool/src/util/customization/interfaces.ts

```diff
@@ -45,20 +63,18 @@ export function mergeProperties(
 ) {
   const customProperties = customInterface.getProperties();
   for (const customProperty of customProperties) {
-    if (getAnnotation(customProperty) === "Remove") {
+    if (getAnnotation(customProperty)?.type === "remove") {
       /* If the property has a `// @azsdk-remove` comment, we don't need to re-add it */
       continue;
     }
     const propertyName = customProperty.getName();
     const originalProperty = originalInterface.getProperty(propertyName);
 
-    // If the property already exists in the original interface, we'll remove it
     if (originalProperty) {
```

Comment:
I think users may be confused if they try to use the `rename` annotation on property declarations, as it doesn't seem that we intend to support that. This isn't actionable at the moment, but we'll have to consider something like type checking to help with this.

Reply:
yeah, I haven't tried that to see what happens... I only focused on the scenarios that I needed to unblock to get my client library to work. Should be doable but really didn't look into it.